openlibrary/plugins/upstream/account.py

```diff
@@ -731,26 +757,52 @@ def get_works(self, key):
             #works = web.ctx.site.get_many([ ... ])
             raise
 
+class public_my_books(delegate.page):
+    path = "/people/([^/]+)/books"
+
+    def GET(self, username):
+        raise web.seeother('/people/%s/books/want-to-read' % username)
+
+class public_my_books(delegate.page):
+    path = "/people/([^/]+)/books/([a-zA-Z_-]+)"
+
+    def GET(self, username, key='loans'):
+        """check if user's reading log is public"""
+        user_preferences = web.ctx.site.get('/people/%s/preferences' % username)
+        notifications = (user_preferences or {}).get('notifications')
+        if notifications and notifications.get('public_readlog', 'yes') == 'yes':
+            readlog = ReadingLog()
+            works = readlog.get_works(key)
+            return render['account/books'](
+                works, key, reading_log=readlog.reading_log_counts,
+                lists=readlog.lists, username=username)
+        return render['generic'](
+            '<h3>Sorry, this user has choosen not to make their reading log public.</h3>' \
```

Comment:
Should read "Sorry, this user has not chosen to make their reading log public." It's a small difference, but reflects a default stance of privacy.